make spo user get command return current user if no arguments are provided

Vedu1996 opened the pull request:
Fixes https://github.com/pnp/cli-microsoft365/issues/5513

Adam-it commented:
Awesome 👍
We will review it ASAP

milanholemans commented on src/m365/spo/commands/user/user-get.spec.ts:
```diff
@@ -215,6 +215,55 @@ describe(commands.USER_GET, () => {
   });
 
+
+  it('retrieves current user with output option json', async () => {
```
```suggestion
  it('retrieves current logged-in user', async () => {
```

milanholemans commented on src/m365/spo/commands/user/user-get.spec.ts:
```diff
@@ -215,6 +215,55 @@ describe(commands.USER_GET, () => {
+
+  it('retrieves current user with output option json', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf('/_api/web/currentuser') > -1) {
```
Let's specify the entire URL rather than using `indexOf`.

milanholemans commented on src/m365/spo/commands/user/user-get.spec.ts:
```diff
@@ -215,6 +215,55 @@ describe(commands.USER_GET, () => {
+
+    await command.action(logger, {
+      options: {
+        output: 'json',
+        debug: true,
```
Let's remove these two options since they are not really needed here.

milanholemans commented on src/m365/spo/commands/user/user-get.ts:
```diff
@@ -76,7 +76,10 @@ class SpoUserGetCommand extends SpoCommand {
+    });
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
```
Noticed that the verbose message below this line is a bit odd. Could we fix that one as well please?

milanholemans commented on src/m365/spo/commands/user/user-get.ts:
```diff
@@ -95,6 +98,9 @@ class SpoUserGetCommand extends SpoCommand {
+      requestUrl = `${args.options.webUrl}/_api/web/currentuser`;
+    }
 
     const requestOptions: any = {
```
```suggestion
    const requestOptions: CliRequestOptions = {
```

milanholemans reviewed (changes requested):
Hi @Vedu1996 great start so far.
I've made a few comments along the way, could you have a look at them, please?

Also, let's add an example to the docs so people know how they can use this enhancement.

milanholemans commented:
Hi, @Vedu1996 seems like this PR has some merge conflicts. Could you rebase it with the latest `main` please? When you are done, you can hit the 'Ready for review' button.
Thanks!

Vedu1996 commented:
> Hi, @Vedu1996 seems like this PR has some merge conflicts. Could you rebase it with the latest `main` please? When you are done, you can hit the 'Ready for review' button. Thanks!
> 
> P.S. I also noticed you forgot to add the example to the docs page.

Hey @milanholemans , I've rebased. Added an example in docs as well. Thanks!

milanholemans commented:
> Hey @milanholemans , I've rebased. Added an example in docs as well. Thanks!

Thanks, will do another approval round soon!

Vedu1996 commented:
> Hi @Vedu1996, seems like the code coverage of the file didn't succeed. Could you have a look at it?
<img width="1057" alt="Screenshot 2023-09-26 at 1 22 35 PM" src="https://github.com/pnp/cli-microsoft365/assets/28388096/ae5315a0-e971-49c0-a38f-8683bef3b3e1">
I ran the coverage locally, and for some reason it is failing in fsUtil.ts, and then the other files are not tested for coverage. Could you let me know how can I fix it please?

milanholemans commented:
> > Hi @Vedu1996, seems like the code coverage of the file didn't succeed. Could you have a look at it?
> 
> <img alt="Screenshot 2023-09-26 at 1 22 35 PM" width="1057" src="https://user-images.githubusercontent.com/28388096/270579065-ae5315a0-e971-49c0-a38f-8683bef3b3e1.png"> I ran the coverage locally, and for some reason it is failing in fsUtil.ts, and then the other files are not tested for coverage. Could you let me know how can I fix it please?

Yeah, you can totally ignore the fsutils. Just pay attention to the files you modified.
Are you sure the changes here in the cloud are in sync with the changes locally on your PC? It seems like we are missing a test where we mock the `currentuser` endpoint.
Did you build your code before running the tests locally?

Vedu1996 commented:
> > > Hi @Vedu1996, seems like the code coverage of the file didn't succeed. Could you have a look at it?
> > 
> > 
> > <img alt="Screenshot 2023-09-26 at 1 22 35 PM" width="1057" src="https://user-images.githubusercontent.com/28388096/270579065-ae5315a0-e971-49c0-a38f-8683bef3b3e1.png"> I ran the coverage locally, and for some reason it is failing in fsUtil.ts, and then the other files are not tested for coverage. Could you let me know how can I fix it please?
> 
> Yeah, you can totally ignore the fsutils. Just pay attention to the files you modified. Are you sure the changes here in the cloud are in sync with the changes locally on your PC? It seems like we are missing a test where we mock the `currentuser` endpoint. Did you build your code before running the tests locally?

I understand, but that is the only file that runs. I assume it runs the coverage, sees that fsUtils has failed and stopes running it altogether, so I cant figure out why coverage is failing. I guess I could look at the GH actions log, but this might be something we want to look at?

Vedu1996 commented:
> > > > Hi @Vedu1996, seems like the code coverage of the file didn't succeed. Could you have a look at it?
> > > 
> > > 
> > > <img alt="Screenshot 2023-09-26 at 1 22 35 PM" width="1057" src="https://user-images.githubusercontent.com/28388096/270579065-ae5315a0-e971-49c0-a38f-8683bef3b3e1.png"> I ran the coverage locally, and for some reason it is failing in fsUtil.ts, and then the other files are not tested for coverage. Could you let me know how can I fix it please?
> > 
> > 
> > Yeah, you can totally ignore the fsutils. Just pay attention to the files you modified. Are you sure the changes here in the cloud are in sync with the changes locally on your PC? It seems like we are missing a test where we mock the `currentuser` endpoint. Did you build your code before running the tests locally?
> 
> I understand, but that is the only file that runs. I assume it runs the coverage, sees that fsUtils has failed and stopes running it altogether, so I cant figure out why coverage is failing. I guess I could look at the GH actions log, but this might be something we want to look at?

I'll build the app and have a look, thanks

Vedu1996 commented:
>  It seems like we are missing a test where we mock the `currentuser` endpoint. Did you build your code before running the tests locally?
My bad, didnt run the build before testing. The missing test was actually because I messed up my rebase, I fixed it now. 
Sorry, and thanks :)

Vedu1996 commented:
Hey, is there something I'm missing on my side? Just checking because I just noticed this PR hasnt been merged yet :)

Jwaegebaert commented:
Hey @Vedu1996, pardon for the wait. We currently have a heavy load of PRs to process so we'll try our best to process this one as soon as we can.

Vedu1996 commented:
Hey, no worries, I was concerned that it there was something wrong with my
pr

On Sun, 8 Oct, 2023, 21:01 Jasey Waegebaert, ***@***.***>
wrote:

> Hey @Vedu1996 <https://github.com/Vedu1996>, pardon for the wait. We
> currently have a heavy load of PRs to process so we'll try our best to
> process this one as soon as we can.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pnp/cli-microsoft365/pull/5514#issuecomment-1752067100>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGYSWADICTSEGWPEQFC63WLX6LBMLAVCNFSM6AAAAAA5EA6KICVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJSGA3DOMJQGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

milanholemans commented:
Hi @Vedu1996, like Jasey mentioned, we had quite a hectic period because of our new major release. Additionally, I haven't had much time in recent weeks. I will try to take a look at it soon.

milanholemans commented on docs/docs/cmd/spo/user/user-get.mdx:
````diff
@@ -49,6 +49,11 @@ Get user by login name for a web
 ```sh
 m365 spo user get --webUrl https://contoso.sharepoint.com/sites/project-x --loginName "i:0#.f|membership|john.doe@mytenant.onmicrosoft.com"
 ```
+Get user logged-in currently
````
```suggestion
Get the currently logged-in user
```

milanholemans commented:
Hi @Vedu1996 I'm currently checking your latest commit.
In the meantime, could you give me your name and surname? That way we can add you to our [team page](https://pnp.github.io/cli-microsoft365/about/team).

milanholemans reviewed (approved):
Great work! Looks good to me! Thank you for your contribution!

I rephrased one small thing while merging.

milanholemans commented:
Thank you very much for this enhancement @Vedu1996! It was a pleasure working with you and we hope to see you again on this project.
You're now officially part of the contributors' team, welcome! https://pnp.github.io/cli-microsoft365/about/team

Vedu1996 commented:
Hey @milanholemans , thanks a lot!
